packages/cli/src/NodeTypes.ts

```diff
@@ -16,12 +16,8 @@ import type { Dirent } from 'fs';
 
 @Service()
 export class NodeTypes implements INodeTypes {
-	constructor(private nodesAndCredentials: LoadNodesAndCredentials) {}
-
-	init() {
-		// Some nodeTypes need to get special parameters applied like the
-		// polling nodes the polling times
-		this.applySpecialNodeParameters();
+	constructor(private nodesAndCredentials: LoadNodesAndCredentials) {
+		nodesAndCredentials.addPostProcessor(async () => this.applySpecialNodeParameters());
```

Comment:
We're expecting the caller to make a sync function async just so that it matches the signature? Is this preferable to `addPostProcessor` knowing how to work with both function kinds, or at least promisifying internally?

Reply:
all sync functions are awaitable, so this won't cause any issues.